src/app/core/rpc/address-rpc.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable } from '@angular/core';
+import { Log } from 'ng2-logger';
+import { Observer } from 'rxjs/Observer';
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/map';
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/observable/throw';
+
+import { RPCService } from './rpc.service';
+import { AddressCount } from './models/address-count.model';
+import { Address } from './models/address.model';
+import { Http } from '@angular/http';
+import {ElectronService} from 'ngx-electron';
+
+@Injectable()
+export class AddressRpcService {
+
+  log: any = Log.create('address-rpc.service');
+
+  private API_URL: string = '';
+
+  /**
+   *
+   * @param http
+   * @param rpc
+   */
+  constructor(private http: Http, public rpc: RPCService) {
+    this.API_URL = this.rpc.getApiUrl();
+  }
+
+  public getAddressCount(): Observable<AddressCount> {
+    const postData = this.rpc.getPostData('filteraddresses', [-1]);
+    const options = this.rpc.getRequestOptions();
+    return this.http
+      .post(this.API_URL, postData, options )
+      .map(response => {
+        const addressCount = response.json().result;
+        this.log.d('getAddressCount, addressCount:', addressCount);
+        return new AddressCount(addressCount);
+      })
+      .catch(this.handleError);
+  }
+
+  public findAddresses(params: Array<any>): Observable<Address[]> {
```

Comment:
I'm not exactly sure if this would work as intended

```
ngOnInit() {
  // get initial addresses using default settings
    this.addressesSubscription = this.addressService
      .findAddresses(this.getPaginationParams(this.addressType, this.currentPage))
      .subscribe(
        (addresses) => {
          this.addresses = addresses;
        }
      );
```

The function returns a _new_ observable each time findAddresses is called, to which you subscribe. When an update occurs to the addresses, it would have to re-call findAddresses (currently only in ngOnInit and onPageChanged) -> address table only updates when you change the page?

I think having _one_ public observables and updating that one would avoid this situation and potentially avoids having to re-subscribe onPageChanged?

There is currently no polling for addresses so the address table doesn't update, I did add it into #24, see specialPoll()